### PR TITLE
compute: write unified introspection updates to storage

### PR DESCRIPTION
**Pull request by teskje**
This PR wraps up the unified compute introspection MVP by adding the missing code to write updates from introspection subscribes to storage, as well as a first unified relation, `mz_compute_error_counts_raw_unified`.

### Motivation

  * This PR adds a known-desirable feature.

Closes https://github.com/MaterializeInc/materialize/issues/26730

### Tips for reviewer

Commits can be reviewed independently.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/A

### CODE REVIEW

**Comment by jkosh44** [comment]
LGTM from a SQL council point of view.

File: src/environmentd/tests/server.rs [hunk 1]
```diff
@@ -1045,13 +1050,11 @@ fn test_closing_connection_cancels_dataflows(query: String) {
...
             let count: i64 = client
-                .query_one(
-                    "SELECT count(*) FROM mz_introspection.mz_dataflow_operators",
-                    &[],
-                )
+                .query_one(DATAFLOW_QUERY, &[])
                 .map_err(|_| ())
                 .unwrap()
                 .get(0);
+            println!("(1) count={count}");
```
Comment (maddyblue) [inline src/environmentd/tests/server.rs:1057]: this print and below can be removed

**Review by maddyblue — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
I triggered a nightly build: https://buildkite.com/materialize/nightly/builds/8308

File: src/environmentd/tests/server.rs (hunk 1 above)
Comment (teskje) [reply src/environmentd/tests/server.rs:1057]: Ups, good catch!

**Comment by teskje** [comment]
TFTRs!

**Pull request merged**